### PR TITLE
[SPARK-30207][SQL][DOCS] Enhance the SQL NULL Semantics document

### PR DESCRIPTION
**Pull request by xuanyuanking**
### What changes were proposed in this pull request?
Enhancement of the SQL NULL Semantics document: sql-ref-null-semantics.html.

### Why are the changes needed?
Clarify the behavior of `UNKNOWN` for both `EXIST` and `IN` operation.

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
Doc changes only.

### CODE REVIEW

**Comment by xuanyuanking** [comment]
cc @gatorsmile @cloud-fan

**Comment by dongjoon-hyun** [comment]
Could you replace `UNKNOW` with `UNKNOWN` in the PR description?

File: docs/sql-ref-null-semantics.md [hunk 1]
```diff
@@ -610,7 +610,8 @@ is a non-membership condition and returns TRUE when no rows or zero rows are
 returned from the subquery.
```
Comment (dongjoon-hyun) [inline docs/sql-ref-null-semantics.md]: It's not related to this PR, but could you fix `In otherwords` to `In other words` at line 607 while you are touching this section?

File: docs/sql-ref-null-semantics.md [hunk 2]
```diff
@@ -668,6 +669,10 @@ To summarize, below are the rules for computing the result of an `IN` expression
   list does not contain NULL values
 - UNKNOWN is returned when the value is `NULL`, or the non-NULL value is not found in the list
   and the list contains at least one `NULL` value
+
+NOT IN always returns UNKNOWN when the list contains NULL, regardless of the input value.
```
Comment (dongjoon-hyun) [inline docs/sql-ref-null-semantics.md]: ``` NULL -> `NULL` ```?

File: docs/sql-ref-null-semantics.md [hunk 3]
```diff
@@ -668,6 +669,10 @@ To summarize, below are the rules for computing the result of an `IN` expression
...
+This is because IN returns UNKNOWN if the value is not in the list, and because NOT UNKNOWN
```
Comment (dongjoon-hyun) [inline docs/sql-ref-null-semantics.md]: ```in the list -> in the list containing `NULL` ```?

**Review by maropu — APPROVED** [review]
LGTM except for the @dongjoon-hyun comments.

File: docs/sql-ref-null-semantics.md (hunk 1 above)
Comment (xuanyuanking) [reply docs/sql-ref-null-semantics.md]: Sure, done in f815452.

File: docs/sql-ref-null-semantics.md (hunk 2 above)
Comment (xuanyuanking) [reply docs/sql-ref-null-semantics.md]: Thanks, done in f815452.

File: docs/sql-ref-null-semantics.md (hunk 3 above)
Comment (xuanyuanking) [reply docs/sql-ref-null-semantics.md]: Thanks, done in f815452.

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by xuanyuanking** [comment]
Thanks!

**Pull request closed without merging**